File: ABC/ABC335/D/main.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> dx = {1, 0, -1, 0};
  vector<int> dy = {0, 1, 0, -1};

  vector res(n, vector<int>(n, -1));
  res[0][0] = 1;

  int x = 0, y = 0, d = 0;
  for (int i = 2; i < n * n; ++i) {
    int nx = x + dx[d];
    int ny = y + dy[d];
    if (nx < 0 || nx >= n || ny < 0 || ny >= n || res[nx][ny] != -1) {
      d = (d + 1) % 4;
      nx = x + dx[d];
      ny = y + dy[d];
    }
    res[x = nx][y = ny] = i;
  }

  for (int i = 0; i < n; ++i) {
    for (int j = 0; j < n; ++j) {
      if (i == n / 2 && j == n / 2) {
        cout << "T ";
      } else {
        cout << res[i][j] << " ";
      }
    }
    cout << endl;
  }

  return 0;
}

File: ABC/ABC335/D/main.py
n = int(input())
res = [[0] * n for _ in range(n)]
res[n // 2][n // 2] = 'T'
res[0][0] = 1

dx = [1, 0, -1, 0]
dy = [0, 1, 0, -1]

x, y, d = 0, 0, 0
for i in range(2, n * n):
    nx = x + dx[d]
    ny = y + dy[d]
    if nx < 0 or nx >= n or ny < 0 or ny >= n or res[nx][ny] != 0:
        d = (d + 1) % 4
        nx = x + dx[d]
        ny = y + dy[d]
    res[nx][ny] = i
    x = nx
    y = ny

for r in res:
    print(*r)